infra/main.tf

```diff
@@ -0,0 +1,190 @@
+  name = "internal"
+}
+
+data "google_compute_subnetwork" "internal_default_region" {
+  name = "internal"
+  region = var.gcp_region
+  depends_on = [google_compute_network.internal]
+}
+
+resource "google_container_cluster" "vdc" {
+  name = "vdc"
+  location = var.gcp_zone
+  network = google_compute_network.internal.name
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  master_auth {
+    username = ""
+    password = ""
+
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+}
+
+resource "google_container_node_pool" "vdc_preemptible_pool" {
+  name       = "preemptible-pool"
+  location   = var.gcp_zone
+  cluster    = google_container_cluster.vdc.name
+
+  autoscaling {
+    min_node_count = 0
+    max_node_count = 200
+  }
+
+  node_config {
+    preemptible  = true
+    machine_type = "n1-standard-2"
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform"
+    ]
+  }
+}
+
+resource "google_container_node_pool" "vdc_nonpreemptible_pool" {
+  name       = "nonpreemptible-pool"
+  location   = var.gcp_zone
+  cluster    = google_container_cluster.vdc.name
+
+  autoscaling {
+    min_node_count = 0
+    max_node_count = 200
+  }
+
+  node_config {
+    preemptible  = false
+    machine_type = "n1-standard-2"
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform"
+    ]
+  }
+}
+
+resource "google_compute_global_address" "db_ip_address" {
+  name          = "db-ip-address"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network = google_compute_network.internal.id
+}
+
+resource "google_service_networking_connection" "private_vpc_connection" {
+  network = google_compute_network.internal.id
+  service = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.db_ip_address.name]
+}
+
+resource "random_id" "db_name_suffix" {
+  byte_length = 4
+}
+
+resource "google_sql_database_instance" "db" {
+  name             = "db-${random_id.db_name_suffix.hex}"
+  database_version = "MYSQL_5_7"
+  region           = var.gcp_region
+
+  depends_on = [google_service_networking_connection.private_vpc_connection]
+
+  settings {
+    # Second-generation instance tiers are based on the machine
+    # type. See argument reference below.
+    tier = "db-n1-standard-1"
+
+    ip_configuration {
+      ipv4_enabled    = false
+      private_network = google_compute_network.internal.id
+      require_ssl = true
+    }
+  }
+}
+
+# FIXME rename this to gateway
+resource "google_compute_address" "gateway" {
+  name = "gateway"
```

Comment:
seems odd to be explicit about region in internal_gateway but implicit here

Reply:
I now set the region here.  Currently all our IP addresses are regional.